test/test_xla_sharding.py

```diff
@@ -104,13 +121,43 @@ def test_deep_copy(self):
         torch_xla._XLAC._get_xla_sharding_spec(xt),
         torch_xla._XLAC._get_xla_sharding_spec(xt2))
 
+  def test_clone(self):
+    xt = torch.randn(2, 4, 8, 16).to(xm.xla_device())
+    xs.mark_sharding(xt, self._get_mesh((1, 1, 1, self.n_devices)),
+                     (0, 1, 2, 3))
+    xt2 = xt.clone()
+    self.assertEqual(
+        torch_xla._XLAC._get_xla_sharding_spec(xt),
+        torch_xla._XLAC._get_xla_sharding_spec(xt2))
```

Comment:
nit, this won't check the case where `clone` will incorrectly remove the sharding_spec from origional tensor and the new tensor. Maybe consider adding another check to save the old sharding_spec and make sure it does not change after `clone`

Reply:
Sounds good, let me extend this.